harmonica/forward/tesseroid_layer.py

```diff
@@ -0,0 +1,353 @@
+        mask = self._get_nonans_mask(property_name=density_name)
+        # Select only the boundaries and density elements for masked tesseroid
+        boundaries = boundaries[mask.ravel()]
+        density = density[mask]
+        # Return gravity field of tesserids
+        return tesseroid_gravity(
+            coordinates,
+            tesseroids=boundaries,
+            density=density,
+            field=field,
+            **kwargs,
+        )
+
+    def _get_nonans_mask(self, property_name=None):
+        """
+        Build a mask for tesseroid with no nans on top, bottom or a property
+
+        Parameters
+        ----------
+        mask : 2d-array
+            Array of bools that can be used as a mask for selecting tesseroids
+            with no nans on top boundaries, bottom boundaries ans the passed
+            property.
+        """
+        # Mask the tesseroid that contains no nans on top and bottom boundaries
+        mask = np.logical_and(
+            np.logical_not(np.isnan(self._obj.top.values)),
+            np.logical_not(np.isnan(self._obj.bottom.values)),
+        )
+        # Mask the tesseroids that contains nans on the selected property
+        if property_name is not None:
+            mask_property = np.logical_not(np.isnan(self._obj[property_name].values))
+            # Warn if a nan is found within the masked property
+            if not mask_property[mask].all():
+                warnings.warn(
+                    'Found missing values in "{}" property '.format(property_name)
+                    + "of the tesseroid layer. "
+                    + "The tesseroids with nan as "
+                    + '"{}" will be ignored.'.format(property_name)
+                )
+            mask = np.logical_and(mask, mask_property)
+            return mask
+
+    def _to_tesseroids(self):
+        """
+        Return the boundaries of each tesseroid of the layer
+
+        Returns
+        -------
+        tesseroids : 2d-array
+            Array containing the boundaries of each tesseroid of the layer.
+            Each row contains the boundaries of each tesseroid in the following
+            order: ``longitude_min``, ``longitude_max``, ``latitude_min``,
+            ``latitude_max``, ``bottom``, ``top``.
+        """
+        longitude, latitude = np.meshgrid(
+            self._obj.longitude.values, self._obj.latitude.values
+        )
+        (
+            longitude_min,
+            longitude_max,
+            latitude_min,
+            latitude_max,
+        ) = self._get_tesseroid_horizontal_boundaries(
+            longitude.ravel(), latitude.ravel()
+        )
+        bottom = self._obj.bottom.values.ravel()
+        top = self._obj.top.values.ravel()
+        tesseroids = np.vstack(
+            (longitude_min, longitude_max, latitude_min, latitude_max, bottom, top)
+        ).T
+        return tesseroids
+
+    def _get_tesseroid_horizontal_boundaries(self, latitude, longitude):
```

Comment:
Here is the buggy line giving that nasty zero division error you showed me today.
```suggestion
    def _get_tesseroid_horizontal_boundaries(self, longitude, latitude):
```